modules/internal/ChapelRange.chpl

```diff
@@ -144,7 +144,8 @@ module ChapelRange {
   
     var _low: idxType = 1;                         // lower bound
     var _high: idxType = 0;                        // upper bound
-    var _stride: strType = 1;                      // signed stride of range
+    var _stride: if stridable then strType else void =
+                   if stridable then 1 else _void; // signed stride of range
     var _alignment: idxType = 0;                   // alignment
     var _aligned : bool = false;
```

Comment:
Currently an unstridable range can have alignment - though I'm not sure it's possible to make it do anything.  I guessed that it would affect the new range that gets created `by` the by operator, but it doesn't appear to.

```
var r = 1..10 align 2;
writeln(r.aligned); // true
writeln(r.alignment); // 2
writeln(r.stridable); // false
writeln((r by 2).alignment); // 1
```

Making them `void` and disallowing this seems fine to me.